pallets/pallet-asset-switch/README.md

```diff
@@ -0,0 +1,129 @@
+
+## Storage
+
+The pallet has a single `SwitchPair` storage value that contains a `Option<SwitchPairInfo>`.
+If unset, no switch pair is configured hence no switch can happen.
+When set and its status is `Running`, switches are enabled in both directions.
+
+## Events
+
+The pallet generates the following events:
+
+- `SwitchPairCreated`: when a new switch pair is created by the required origin, e.g., governance.
+- `SwitchPairRemoved`: when a switch pair is removed by the root origin.
+- `SwitchPairResumed`: when a switch pair has (re-)enabled local to remote asset switches.
+- `SwitchPairPaused`: when a switch pair has been paused.
+- `SwitchPairFeeUpdated`: when the XCM fee for the switch transfer has been updated.
+- `LocalToRemoteSwitchExecuted`: when a switch of some local tokens for the remote asset has taken place.
+- `RemoteToLocalSwitchExecuted`: when a switch of some remote assets for the local tokens has taken place.
+
+## Calls
+
+1. `pub fn set_switch_pair(origin: OriginFor<T>, reserve_location: Box<VersionedMultiLocation>, remote_asset_id: Box<VersionedAssetId>, remote_fee: Box<VersionedMultiAsset>, total_issuance: u128, circulating_supply: u128) -> DispatchResult`: Set a new switch pair between the local currency and the specified `remote_asset_id` on the `reserve_location`. The specified `total_issuance` includes both the `circulating_supply` (i.e., the remote asset amount that the chain does not control on the `reserve_location`) and the locked supply under the control of the chain's sovereign account on the `reserve_location`. For this reason, the value of `total_issuance` must be at least as large as `circulating_supply`. It is possible for `circulating_supply` to be `0`, in which case it means this chain controls all of the `total_issuance` of the remote asset, which can be obtained by locking a corresponding amount of local tokens via the `switch` call below. Furthermore, the pallet calculates the account that will hold the local tokens locked in exchange for remote tokens: this account is based on the pallet runtime name as returned by the `PalletInfoAccess` trait, and the value of `remote_asset_id`. The generated account must already have a balance of at least `circulating_supply`, ensuring that there are enough local tokens locked to satisfy all the requests to exchange the remote asset for local tokens. The balance of such account can be increased with a simple transfer, after obtaining the to-be-created switch pair pool account by interacting with the [asset-switch runtime API][asset-switch-runtime-api]: this requirement is bypassed with the `force_set_switch_pair` call. Only `SwitchOrigin` can call this, and in most cases it will most likely be a governance-based origin such as the one provided by referenda or collectives with high privileges.
```

Comment:
```suggestion
1. `pub fn set_switch_pair(origin: OriginFor<T>, reserve_location: Box<VersionedMultiLocation>, remote_asset_id: Box<VersionedAssetId>, remote_fee: Box<VersionedMultiAsset>, total_issuance: u128, circulating_supply: u128) -> DispatchResult`: Set a new switch pair between the local currency and the specified `remote_asset_id` on the `reserve_location`. The specified `total_issuance` includes both the `circulating_supply` (i.e., the remote asset amount that the chain does not control on the `reserve_location`) and the locked supply under the control of the chain's sovereign account on the `reserve_location`. For this reason, the value of `total_issuance` must be at least as large as `circulating_supply`. It is possible for `circulating_supply` to be `0`, in which case it means this chain controls all of the `total_issuance` of the remote asset, which can be obtained by locking a corresponding amount of local tokens via the `switch` call below.

Furthermore, the pallet calculates the account that will hold the local tokens locked in exchange for remote tokens. This account is based on the pallet runtime name as returned by the `PalletInfoAccess` trait and the value of `remote_asset_id`. The generated account must already have a balance of at least `circulating_supply`, ensuring enough local tokens are locked to satisfy all requests to exchange the remote asset for local tokens. The balance of such an account can be increased with a simple transfer after obtaining the to-be-created switch pair pool account by interacting with the [asset-switch runtime API][asset-switch-runtime-api].

This requirement can be bypassed with the `force_set_switch_pair` call. Only `SwitchOrigin` can call this, and in most cases it will most likely be a governance-based origin such as the one provided by referenda or collectives with high privileges.
```

Reply:
This is a bullet point, I don't think we can add whitespaces without breaking the list, can we?

Reply:
Ah yes, it needs to be intended, though to be fair, this is very long for a bullet point, but a bit of whitespace will help.

Reply:
I think that's right, GitHub makes it hard to suggest indents, if not, I'll just submit the change directly.

Reply:
Ok I'll merge this suggestion in the PR I opened.